### PR TITLE
Add Metrics Histogram example in folder examples/mp-metrics-histogram

### PR DESCRIPTION
**Pull request by mpredli01**
Hello everyone:

I added the Metrics Histogram example as described in TOMEE-2287.

Mike.

### CODE REVIEW

**Comment by josehenriqueventura** [comment]
Taking the opportunity to help to review PR as David mentioned earlier, I would like to suggest some improvements.

**Comment by josehenriqueventura** [comment]
Also when compiling the parent project "example/pom.xml" your mp-metrics-histogram project won't be compiled because is missing an entry of mp-metrics-histogram there.
I suggest you rebase your branch with the latest version and add the project after the tag which describes the link below.

https://github.com/apache/tomee/blob/e504c4df0cd847172ccec019a958c63f8b58a0cd/examples/pom.xml#L179

File: examples/mp-metrics-histogram/pom.xml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
```
Comment (brunobat) [inline examples/mp-metrics-histogram/pom.xml]: Please remove the parent dependency. For details please check: https://github.com/apache/tomee/blob/master/examples/README.adoc

File: examples/mp-metrics-histogram/url-output.txt [hunk 2]
```diff
@@ -0,0 +1,164 @@
+Last login: Tue Dec  4 18:42:11 on ttys001
```
Comment (brunobat) [inline examples/mp-metrics-histogram/url-output.txt]: Is this file really needed?

File: examples/mp-opentracing-traced/pom.xml [hunk 3]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (brunobat) [inline examples/mp-opentracing-traced/pom.xml]: Is this file really needed? It seems to be related to a different example.

File: examples/mp-opentracing-traced/src/test/resources/arquillian.xml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
```
Comment (brunobat) [inline examples/mp-opentracing-traced/src/test/resources/arquillian.xml]: Is this file really needed? It seems to be related to a different example.

**Comment by mpredli01** [comment]
Hi Bruno...

Thanks for catching the url-output.txt file.  It was obviously something I was using while testing my example...

What's the best course of action - just delete it from my forked and local repos?  Or delete it from my local repo and do another push.  I need to edit the pom.xml file anyway.

With regards to the new folder, it's the start of my new example.  I forgot to exclude it from my latest push.  Again, what's the best course of action.  It will ultimate get updated anyway, so can it come for the ride now?

Please advise...thanks!

Mike.

**Comment by brunobat** [comment]
Hi @mpredli01,
You can delete those files on your local branch and push the changes. 
That should be enough.
Cheers!

**Comment by mpredli01** [comment]
Hi @brunobat:

K, thanks!  I will do this during lunch today...

Mike.

File: examples/mp-metrics-histogram/src/main/java/org/superbiz/histogram/WeatherService.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        displayName = "Histogram of Recent New York Temperatures")
+    private Histogram histogram;
+
+    @Path("/histogram")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Histogram getTemperatures() {
+        Metadata metadata = new Metadata("temperatures", MetricType.HISTOGRAM, "degrees F");
+        metadata.setDescription("A histogram of recent New York temperatures.");
+        final int[] RECENT_NEW_YORK_TEMPS = { 46, 45, 50, 46, 45, 27, 30, 48, 55, 54, 45, 41, 45, 43, 46 };
```
[the remark this answers is not in the document]
Comment (mpredli01) [reply examples/mp-metrics-histogram/src/main/java/org/superbiz/histogram/WeatherService.java]: Hi Jose...

Thanks for the suggestion.  I already removed the <parent></parent> section in the pom.xml file, but I will make the other changes tonight and push to my repo.

**Comment by mpredli01** [comment]
Hello, everyone:

Jose's suggestions have been implemented and pushed to the repo...

Mike.
